node/network/collator-protocol/src/collator_side.rs

```diff
@@ -638,8 +654,8 @@ where
 			handle_peer_view_change(ctx, state, peer_id, view).await?;
 		}
 		PeerDisconnected(peer_id) => {
-			state.known_validators.retain(|_, v| *v != peer_id);
 			state.peer_views.remove(&peer_id);
+			state.declared_at.remove(&peer_id);
```

Comment:
do we also need to remove it from `peer_ids` mapping in `our_validator_groups`?

Reply:
No it is fine to stay in there. `our_validator_groups` is mainly used to track `PeerId` -> `ValidatorId` mapping and will be cleaned up when the corresponding leaf is closed.